libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/FileUtils.java

```diff
@@ -58,4 +73,127 @@ public static String getFileNameFromPath(String filePath) {
         }
         return filename;
     }
+
+    /**
+     * This solution is based on https://stackoverflow.com/a/53021624
+     * In certain cases we cannot load a file from the disk so we have to cache it instead using streams.
+     * The helper methods are copied from - https://github.com/coltoscosmin/FileUtils/blob/master/FileUtils.java
+     *
+     * @param context The context.
+     * @param uri The Uri to query.
+     */
+    static String cacheFile(Context context, Uri uri) {
+        String fileName = getFileName(context, uri);
+        File cacheDir = getDocumentCacheDir(context);
```

Comment:
Why not reusing the application cache dir here?  `File cacheDir = context.getCacheDir();` Pretty much what we use in MediaUtils [here](https://github.com/wordpress-mobile/WordPress-Android/blob/bb8f7ee2d4d02c960b6e76beb14f6bedf4d1fc68/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/MediaUtils.java#L196).

Reply:
I think we can. This is creating a nested dir which (I think) is not necessary)